runtime/doc/message.txt

```diff
@@ -836,4 +836,33 @@ The |g<| command can be used to see the last page of previous command output.
 This is especially useful if you accidentally typed <Space> at the hit-enter
 prompt.
 
+
+						*message-function*  >
+
+If the 'msgfunc' option is set, then the user specified function is
+invoked to display messages instead.
+
+The arguments are:
+
+    a:method	"msg_clear", "msg_history_show", or "msg_show"
+    a:kind	specify internal message command.
+    a:chunks	[attr, text] list
```

Comment:
attr -> attr_id

`attr_id` is consistent with how it is described in the "content" item of `:help ui-messages`

More specifically this is a highlight group id from what I've found, thus it could be named `highlight_id` or something along those lines. If that is correct then users are likely to consume those ids in "hl_group" of `nvim_buf_set_extmark` or `nvim_buf_add_highlight`, so it might be worthwhile to mention those functions somewhere.